app/Database/CourseQueries.hs

```diff
@@ -271,3 +273,22 @@ queryGraphs =
         do graphs :: [Entity Graph] <- selectList [] []
            return $ createJSONResponse graphs
 
+-- ========================================================
+
+getLectureTime :: [String] -> IO (Maybe [Time])
+getLectureTime [code, section, session] = runSqlite databasePath $ do 
```

Comment:
Do you mean using (String, String, String) to filter out cases not having 3 Strings (i.e. parsing wrongly)?